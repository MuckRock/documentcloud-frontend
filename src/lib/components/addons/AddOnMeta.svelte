<script lang="ts">
  import type { AddOnListItem } from "@/addons/types";

  import DOMPurify from "isomorphic-dompurify";
  import { _ } from "svelte-i18n";
  import { MarkGithub16 } from "svelte-octicons";

  import Button from "$lib/components/common/Button.svelte";
  import Flex from "@/lib/components/common/Flex.svelte";
  import Metadata from "../common/Metadata.svelte";

  import { ALLOWED_TAGS, ALLOWED_ATTR } from "@/config/config.js";

  export let addon: AddOnListItem;

  $: repo = new URL(addon.repository, "https://github.com/").href;
  $: github_org = addon.repository.split("/")[0];
  $: description = addon.parameters.description
    ? clean(addon.parameters.description)
    : "";
  $: instructions = addon.parameters.instructions
    ? clean(addon.parameters.instructions)
    : "";

  function clean(html: string): string {
    return DOMPurify.sanitize(html, { ALLOWED_TAGS, ALLOWED_ATTR });
  }
</script>

<div class="container">
  <h2 class="name">{addon.name}</h2>
  <div class="description">
    {@html addon.parameters?.description}
  </div>
  {#if instructions}
    <div class="instructions">
      <details>
        <summary>
          {$_("addonDispatchDialog.instructions")}
        </summary>
        {@html instructions}
      </details>
    </div>
  {/if}
  <Flex wrap align="center" justify="between">
    <Metadata key={$_("addonDispatchDialog.createdBy")}>
      <h3>{github_org}</h3>
    </Metadata>
    <Flex>
      <Button ghost mode="primary" href={repo} target="_blank">
        <MarkGithub16 />
        {$_("addonDispatchDialog.viewsource")}
      </Button>
    </Flex>
  </Flex>
</div>

<style>
  h2 {
    margin-bottom: 1rem;
    font-size: var(--font-xl);
  }

  h3 {
    font-size: var(--font-l);
  }

  h2,
  h3 {
    font-weight: var(--font-semibold);
  }

  .description,
  .instructions {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    line-height: 1.5;
    margin-bottom: 1rem;
  }
</style>
