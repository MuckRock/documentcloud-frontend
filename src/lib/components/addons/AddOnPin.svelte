<script lang="ts" context="module">
  import { writable, type Writable } from "svelte/store";
  import type { AddOnListItem } from "$lib/api/types";

  export const pinned: Writable<AddOnListItem[]> = writable([]);
</script>

<script lang="ts">
  import { getCsrfToken } from "$lib/utils/api";
  import { BASE_API_URL } from "@/config/config";
  import Pin from "../common/Pin.svelte";

  export let addon: AddOnListItem;
  export let size = 1;

  $: endpoint = new URL(`/api/addons/${addon.id}/`, BASE_API_URL);

  async function toggle(event) {
    event.preventDefault();

    const csrftoken = getCsrfToken();
    if (!csrftoken) {
      console.error("No CSRF token found");
      return;
    }
    const options: RequestInit = {
      credentials: "include",
      method: "PATCH", // this component can only update whether an addon is active or not
      headers: { "X-CSRFToken": csrftoken, "Content-type": "application/json" },
    };

    // optimistic update
    addon.active = !addon.active;

    const resp = await fetch(endpoint, {
      ...options,
      body: JSON.stringify({ active: addon.active }),
    }).catch((err) => {
      addon.active = !addon.active;
      return {
        ok: false,
        statusText: String(err),
      };
    });

    if (!resp.ok) {
      // reset active state
      addon.active = !addon.active;
      console.error(`Problem updating add-on: ${resp.statusText}`);
    }

    // now that we've updated, set $pinned
    $pinned = addon.active
      ? [...$pinned, addon]
      : $pinned.filter((a) => a.id !== addon.id);
  }
</script>

<Pin
  active={addon.active}
  on:click={toggle}
  {size}
  --fill={addon.active ? "var(--orange)" : "var(--gray-3)"}
/>
