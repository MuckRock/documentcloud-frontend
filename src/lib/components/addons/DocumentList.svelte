<script lang="ts">
  import { _ } from "svelte-i18n";
  import { Hourglass24, SidebarExpand16 } from "svelte-octicons";

  import type { DocumentResults, Maybe } from "$lib/api/types";
  import Flex from "../common/Flex.svelte";
  import Empty from "../common/Empty.svelte";
  import Button from "../common/Button.svelte";
  import PageToolbar from "../toolbars/PageToolbar.svelte";
  import ResultsList, {
    selected,
    selectedIds,
    total,
    visible,
  } from "../documents/ResultsList.svelte";
  import Search from "../forms/Search.svelte";
  import ContentLayout from "../layouts/ContentLayout.svelte";
  import { sidebars } from "../layouts/Sidebar.svelte";
  import { remToPx } from "@/lib/utils/layout";

  export let search: Promise<Maybe<DocumentResults>>;
  export let query: string;

  function selectAll(e) {
    if (e.target.checked) {
      $selectedIds = [...$visible.keys()];
    } else {
      $selectedIds = [];
    }
  }

  let clientWidth: number;
</script>

<div bind:clientWidth>
  <ContentLayout>
    <Flex slot="header">
      {#if $sidebars["navigation"] === false && clientWidth > remToPx(30)}
        <div class="toolbar w-auto">
          <Button
            ghost
            minW={false}
            on:click={() => ($sidebars["navigation"] = true)}
          >
            <span class="flipV">
              <SidebarExpand16 />
            </span>
          </Button>
        </div>
      {/if}
      <PageToolbar>
        <Search name="q" {query} slot="center" />
      </PageToolbar>
    </Flex>
    <svelte:fragment>
      {#await search}
        <Empty icon={Hourglass24}>{$_("common.loading")}</Empty>
      {:then search}
        <ResultsList
          results={search?.results}
          next={search?.next}
          count={search?.count}
          auto
        />
      {/await}
    </svelte:fragment>

    <PageToolbar slot="footer">
      <label slot="left" class="select-all">
        <input
          type="checkbox"
          name="select_all"
          checked={$selected.length === $visible.size}
          indeterminate={$selected.length > 0 &&
            $selected.length < $visible.size}
          on:change={selectAll}
        />
        {#if $selected.length > 0}
          {$selected.length.toLocaleString()} {$_("inputs.selected")}
        {:else}
          {$_("inputs.selectAll")}
        {/if}
      </label>

      <svelte:fragment slot="right">
        {#if $visible && $total}
          {$_("inputs.resultsCount", {
            values: { n: $visible.size, total: $total },
          })}
        {/if}
      </svelte:fragment>
    </PageToolbar>
  </ContentLayout>
</div>
