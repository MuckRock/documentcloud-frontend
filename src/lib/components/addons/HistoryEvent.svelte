<script lang="ts">
  import type { Run } from "$lib/api/types";
  import { _ } from "svelte-i18n";
  import {
    CheckCircle24,
    Paperclip16,
    Hourglass24,
    Alert24,
    Question24,
    Sync24,
  } from "svelte-octicons";
  import Action from "$lib/components/common/Action.svelte";
  import Price from "$lib/components/premium-credits/Price.svelte";

  export let run: Run;

  $: ranAt = new Date(run.created_at);
</script>

<div class="addon-run" id="run-{run.uuid}">
  <div class="status">
    {#if run.status === "success"}
      <span class="success icon" title="Success"><CheckCircle24 /></span>
    {:else if run.status === "failure"}
      <span class="failure icon" title="Failure"><Alert24 /></span>
    {:else if run.status === "queued"}
      <span class="queued icon" title="Queued"><Hourglass24 /></span>
    {:else if run.status === "in_progress"}
      <span class="in-progress icon" title="In Progress"><Sync24 /></span>
    {:else}
      <span class="unknown-status icon" title="Unknown Status">
        <Question24 />
      </span>
    {/if}
  </div>
  <div class="info">
    <div class="row">
      <div class="primary-info">
        <p class="name">{run.addon.name}</p>
        {#if run.file_url}
          <a href={run.file_url} download>
            <Action>
              <Paperclip16 />{$_("processing.download")}
            </Action>
          </a>
        {/if}
      </div>
      <time
        class="date"
        datetime={ranAt.toISOString()}
        title={ranAt.toISOString()}
      >
        {ranAt.toLocaleString()}
      </time>
    </div>
    <div class="row">
      {#if run.message}
        <p class="message">{run.message}</p>
      {/if}
      {#if run.credits_spent}
        <p class="price">
          <Price value={run.credits_spent} />
        </p>
      {/if}
    </div>
  </div>
</div>

<style>
  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
  .addon-run {
    margin: 0.5em;
    display: flex;
    align-items: flex-start;
    gap: 1em;
  }
  .status {
    flex: 0 1 auto;
    margin: 0;
  }
  .success.icon {
    fill: var(--green-3);
  }
  .failure.icon {
    fill: var(--caution);
  }
  .queued.icon {
    fill: var(--gray-3);
  }
  .in-progress.icon {
    display: block;
    fill: var(--primary);
    transform-origin: center center;
    animation: spin 2s linear infinite reverse;
    animation-play-state: running;
    & svg {
      display: block;
    }
  }
  .unknown-status.icon {
    fill: var(--gray-3);
  }
  .row {
    display: flex;
  }
  .info {
    flex: 1 1 auto;
  }
  .info .row {
    gap: 1rem;
    align-items: baseline;
    justify-content: space-between;
  }
  .primary-info {
    flex: 1 1 auto;
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    gap: 0.5em;
  }
  .name {
    flex: 0 1 auto;
    font-weight: 600;
  }
  .date {
    flex: 0 1 auto;
    font-size: 0.8em;
    color: var(--gray-5);
  }
  .message {
    flex: 1 1 100%;
    margin: 0.5em 0 0;
    font-size: 0.8em;
    font-style: italic;
  }
  .price {
    margin: 0.5em 0 0;
    font-size: 0.8em;
    align-self: flex-end;
  }
</style>
