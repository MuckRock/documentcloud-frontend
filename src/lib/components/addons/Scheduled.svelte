<script lang="ts">
  import type { Maybe, Nullable, Page, Event } from "$lib/api/types";

  import { _ } from "svelte-i18n";
  import { Clock16, Hourglass24 } from "svelte-octicons";

  import ScheduledEvent from "./ScheduledEvent.svelte";
  import SidebarGroup from "../sidebar/SidebarGroup.svelte";
  import SidebarItem from "../sidebar/SidebarItem.svelte";
  import Paginator from "$lib/components/common/Paginator.svelte";
  import Empty from "../common/Empty.svelte";

  export let events: Event[];
  export let previous: Maybe<Nullable<string>> = undefined;
  export let next: Maybe<Nullable<string>> = undefined;

  export let loading = false;

  // load the next set of results
  async function load(url: URL) {
    loading = true;

    // todo: better error handling
    const res = await fetch(url, { credentials: "include" }).catch(
      console.error,
    );
    if (!res) return console.error("API error");
    if (!res.ok) {
      console.error(res.statusText);
      loading = false;
    }

    const results: Page<Event> = await res.json();

    events = results.results;
    next = results.next;
    previous = results.previous;
    loading = false;
  }
</script>

<SidebarGroup>
  <SidebarItem slot="title">
    <Clock16 slot="start" />
    {$_("addonRuns.scheduled")}
  </SidebarItem>

  {#if loading}
    <Empty icon={Hourglass24}>{$_("common.loading")}</Empty>
  {:else}
    {#each events as event}
      <ScheduledEvent {event} />
    {:else}
      <Empty>
        {$_("common.empty")}
      </Empty>
    {/each}
  {/if}

  {#if previous || next}
    <Paginator
      has_next={Boolean(next)}
      has_previous={Boolean(previous)}
      on:next={(e) => {
        if (next) {
          load(new URL(next));
        }
      }}
      on:previous={(e) => {
        if (previous) {
          load(new URL(previous));
        }
      }}
    />
  {/if}
</SidebarGroup>
