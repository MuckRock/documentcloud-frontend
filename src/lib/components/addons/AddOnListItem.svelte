<script lang="ts">
  import type { AddOnListItem } from "@/addons/types";

  import { _ } from "svelte-i18n";

  import AddOnPin from "@/addons/AddOnPin.svelte";
  import PremiumBadge from "@/premium-credits/PremiumBadge.svelte";

  export let addon: AddOnListItem;

  $: url = `/add-ons/${addon?.repository}/`;
  $: description = addon?.parameters?.description;
  $: author = { name: addon?.repository?.split("/")[0] };
  $: isPremium = addon?.parameters?.categories?.includes("premium") ?? false;
</script>

<a class="addon-link" href={url}>
  <div class="container" id={addon.repository}>
    <div class="row">
      <div class="center-self">
        <AddOnPin {addon} />
      </div>
      <div class="stretch">
        <h3 class="addon-name">{addon.name}</h3>
      </div>
      <div class="metadata">
        {#if author?.name}
          <p class="author">
            <a
              href="http://github.com/{addon.repository}"
              target="_blank"
              rel="noopener noreferrer"
              title={$_("addonBrowserDialog.viewsource")}>{author.name}</a
            >
          </p>
        {/if}
        {#if isPremium}
          <span class="badge" role="status"><PremiumBadge /></span>
        {/if}
      </div>
    </div>
    {#if description}
      <div class="description">{@html description}</div>
    {/if}
  </div>
</a>

<style>
  a {
    color: inherit;
    text-decoration: none;
  }

  .container {
    display: block;
    min-width: 12rem;
    padding: 0.5rem 0.5rem 0.75rem;
    text-align: left;
  }

  .addon-link:hover .container {
    background-color: var(--blue-1);
  }

  .row {
    display: flex;
    align-items: flex-end;
    gap: 0.5rem;
    margin: 0.5rem;
  }

  .badge {
    margin-bottom: -0.25em;
    font-size: 0.8em;
  }

  .metadata {
    display: flex;
    align-items: flex-end;
    gap: 1rem;
    color: var(--gray-4);
  }

  .description {
    margin: 0 0.5em;
    opacity: 0.6;
    font-size: 0.875em;
    line-height: 1.4;
    color: var(--gray-4);
    overflow: hidden;
    -webkit-line-clamp: 4;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    & > * {
      margin-top: 0;
      font-size: 0.875rem;
    }
  }

  .addon-name {
    margin: 0;
    font-weight: 600;
    font-size: var(--font-lg);
  }

  .center-self {
    align-self: center;
  }

  .stretch {
    flex: 1 1 auto;
  }

  .author a:hover {
    opacity: 0.7;
  }

  p {
    margin: 0;
  }
</style>
