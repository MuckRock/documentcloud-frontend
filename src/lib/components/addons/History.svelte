<script lang="ts">
  import type { Maybe, Nullable, Page } from "$lib/api/types";
  import type { Run } from "$lib/api/types";

  import { _ } from "svelte-i18n";
  import { Alert24, History16, History24, Hourglass24 } from "svelte-octicons";

  import HistoryEvent from "./HistoryEvent.svelte";
  import Paginator from "$lib/components/common/Paginator.svelte";
  import SidebarGroup from "../sidebar/SidebarGroup.svelte";
  import SidebarItem from "../sidebar/SidebarItem.svelte";
  import Empty from "../common/Empty.svelte";

  import { getApiResponse } from "$lib/utils/api";

  export let runs: Run[];
  export let previous: Maybe<Nullable<string>> = undefined;
  export let next: Maybe<Nullable<string>> = undefined;

  export let loading = false;

  let error: string = "";

  $: empty = runs.length === 0;

  // load the next set of results
  async function load(url: URL) {
    loading = true;

    const resp = await fetch(url, { credentials: "include" }).catch(
      console.warn,
    );

    const { data: results, error: err } = await getApiResponse<Page<Run>>(resp);

    if (err) {
      error = err.message;
    }

    if (results) {
      runs = results.results;
      next = results.next;
      previous = results.previous;
    }

    loading = false;
  }
</script>

<SidebarGroup>
  <SidebarItem slot="title">
    <History16 slot="start" />
    {$_("addonRuns.previous")}
  </SidebarItem>

  {#if loading}
    <Empty icon={Hourglass24}>Loading past runsâ€¦</Empty>
  {:else if error}
    <Empty icon={Alert24}>
      {error}
    </Empty>
  {:else}
    {#each runs as run}
      <HistoryEvent {run} />
    {:else}
      <Empty icon={History24}>No past runs</Empty>
    {/each}
  {/if}

  {#if !loading && !empty && (previous || next)}
    <Paginator
      has_next={Boolean(next)}
      has_previous={Boolean(previous)}
      on:next={() => {
        if (next) load(new URL(next));
      }}
      on:previous={() => {
        if (previous) load(new URL(previous));
      }}
    />
  {/if}
</SidebarGroup>
