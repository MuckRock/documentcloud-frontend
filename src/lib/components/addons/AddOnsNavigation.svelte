<script lang="ts">
  import { _ } from "svelte-i18n";
  import { Infinity16, StarFill16 } from "svelte-octicons";

  import Pin from "@/common/icons/Pin.svelte";
  import Premium from "@/common/icons/Premium.svelte";
  import SidebarItem from "$lib/components/sidebar/SidebarItem.svelte";
  import Flex from "../common/Flex.svelte";

  import { CATEGORIES } from "$lib/api/addons";

  export let active: string = "all";
</script>

<Flex direction="column" gap={1}>
  <header class="header">
    <h2>{$_("addonBrowserDialog.title")}</h2>
    <p>{$_("addonBrowserDialog.subtitle")}</p>
  </header>
  <Flex direction="column">
    <SidebarItem
      active={active === "all"}
      href="/app/add-ons"
      --hover-background="var(--blue-2)"
    >
      <Infinity16 fill="var(--blue-3)" />
      All
    </SidebarItem>
    <SidebarItem
      active={active === "active"}
      href="/app/add-ons?active=true"
      --hover-background="var(--orange-light)"
    >
      <Pin --fill="var(--orange)" />
      Pinned
    </SidebarItem>
    <SidebarItem
      active={active === "featured"}
      href="/app/add-ons?featured=true"
      --hover-background="var(--yellow-light)"
    >
      <StarFill16 fill="var(--yellow)" />
      Featured
    </SidebarItem>
    <SidebarItem
      active={active === "premium"}
      href="/app/add-ons?premium=true"
      --hover-background="var(--green-light)"
    >
      <Premium --fill="var(--green)" />
      Premium
    </SidebarItem>
  </Flex>
  <Flex direction="column">
    <SidebarItem small --color="var(--gray-4)">Collections</SidebarItem>
    {#each CATEGORIES as [key, label]}
      <SidebarItem
        href={`/app/add-ons?category=${key}`}
        active={active === key}
      >
        <!-- <Hash16 fill="var(--gray-4)" /> -->
        {label}
      </SidebarItem>
    {/each}
  </Flex>
</Flex>
