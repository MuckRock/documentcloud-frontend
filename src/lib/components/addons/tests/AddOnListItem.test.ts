import { test, expect } from "vitest";
import { render } from "@testing-library/svelte";

import AddOnListItem from "../AddOnListItem.svelte";
import { addon, premiumAddon } from "@/test/fixtures/addons";

test("AddOnListItem", () => {
  const result = render(AddOnListItem, { addon });
  expect(result.getByRole("heading")).toHaveTextContent(addon.name);
  expect(result.container).toMatchSnapshot();
  const premium = render(AddOnListItem, { addon: premiumAddon });
  expect(premium.getByRole("status")).toHaveTextContent("Premium");
  expect(premium.container).toMatchSnapshot();
});
