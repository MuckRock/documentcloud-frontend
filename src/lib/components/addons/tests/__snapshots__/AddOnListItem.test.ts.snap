// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddOnListItem 1`] = `
<div>
  <a
    class="addon-link svelte-zccvfh"
    href="/add-ons/MuckRock/documentcloud-bulk-edit-addon/"
  >
    <div
      class="container svelte-zccvfh"
      id="MuckRock/documentcloud-bulk-edit-addon"
    >
      <div
        class="row svelte-zccvfh"
      >
        <div
          class="center-self svelte-zccvfh"
        >
          <div
            style="display: contents; --fill: var(--gray-3);"
          >
            <button
              class="pin svelte-ne3jki"
              style="--padding:0.25em; --border-radius:4px;"
            >
              <svg
                class="svelte-yezalu"
                height="16px"
                viewBox="0 0 16 16"
                width="16px"
              >
                <title>
                  Click to Pin
                </title>
                <path
                  d="M3.00220543,1 C3.00220543,0.446875 3.44891524,0 4.00183577,0 L11.9988785,0 C12.551799,0 12.9985088,0.446875 12.9985088,1 C12.9985088,1.553125 12.551799,2 11.9988785,2 L11.0773442,2 L11.4334625,6.63125 C12.5799136,7.253125 13.4858286,8.29375 13.9169192,9.590625 L13.9481576,9.684375 C14.0512445,9.990625 13.9981391,10.325 13.8107084,10.584375 C13.6232777,10.84375 13.3202648,11 12.9985088,11 L3.00220543,11 C2.68044942,11 2.38056032,10.846875 2.19000578,10.584375 C1.99945125,10.321875 1.94946973,9.9875 2.05255661,9.684375 L2.08379506,9.590625 C2.51488564,8.29375 3.42080063,7.253125 4.56725167,6.63125 L4.92336998,2 L4.00183577,2 C3.44891524,2 3.00220543,1.553125 3.00220543,1 Z M7.00072677,12 L8.99998744,12 L8.99998744,15 C8.99998744,15.553125 8.55327764,16 8.00035711,16 C7.44743658,16 7.00072677,15.553125 7.00072677,15 L7.00072677,12 Z"
                  fill-rule="nonzero"
                />
              </svg>
            </button>
          </div>
        </div>
         
        <div
          class="stretch svelte-zccvfh"
        >
          <h3
            class="addon-name svelte-zccvfh"
          >
            Bulk Edit
          </h3>
        </div>
         
        <div
          class="metadata svelte-zccvfh"
        >
          <p
            class="author svelte-zccvfh"
          >
            <a
              class="svelte-zccvfh"
              href="http://github.com/MuckRock/documentcloud-bulk-edit-addon"
              rel="noopener noreferrer"
              target="_blank"
              title="View Source"
            >
              MuckRock
            </a>
          </p>
           
        </div>
      </div>
       
      <div
        class="description svelte-zccvfh"
      >
        This will update document information for all documents in your current search.  Please be sure you have edit access to these documents before running this add-on.
      </div>
    </div>
  </a>
</div>
`;

exports[`AddOnListItem 2`] = `
<div>
  <a
    class="addon-link svelte-zccvfh"
    href="/add-ons/MuckRock/documentcloud-azure-document-intelligence-ocr-addon/"
  >
    <div
      class="container svelte-zccvfh"
      id="MuckRock/documentcloud-azure-document-intelligence-ocr-addon"
    >
      <div
        class="row svelte-zccvfh"
      >
        <div
          class="center-self svelte-zccvfh"
        >
          <div
            style="display: contents; --fill: var(--gray-3);"
          >
            <button
              class="pin svelte-ne3jki"
              style="--padding:0.25em; --border-radius:4px;"
            >
              <svg
                class="svelte-yezalu"
                height="16px"
                viewBox="0 0 16 16"
                width="16px"
              >
                <title>
                  Click to Pin
                </title>
                <path
                  d="M3.00220543,1 C3.00220543,0.446875 3.44891524,0 4.00183577,0 L11.9988785,0 C12.551799,0 12.9985088,0.446875 12.9985088,1 C12.9985088,1.553125 12.551799,2 11.9988785,2 L11.0773442,2 L11.4334625,6.63125 C12.5799136,7.253125 13.4858286,8.29375 13.9169192,9.590625 L13.9481576,9.684375 C14.0512445,9.990625 13.9981391,10.325 13.8107084,10.584375 C13.6232777,10.84375 13.3202648,11 12.9985088,11 L3.00220543,11 C2.68044942,11 2.38056032,10.846875 2.19000578,10.584375 C1.99945125,10.321875 1.94946973,9.9875 2.05255661,9.684375 L2.08379506,9.590625 C2.51488564,8.29375 3.42080063,7.253125 4.56725167,6.63125 L4.92336998,2 L4.00183577,2 C3.44891524,2 3.00220543,1.553125 3.00220543,1 Z M7.00072677,12 L8.99998744,12 L8.99998744,15 C8.99998744,15.553125 8.55327764,16 8.00035711,16 C7.44743658,16 7.00072677,15.553125 7.00072677,15 L7.00072677,12 Z"
                  fill-rule="nonzero"
                />
              </svg>
            </button>
          </div>
        </div>
         
        <div
          class="stretch svelte-zccvfh"
        >
          <h3
            class="addon-name svelte-zccvfh"
          >
            Azure Document Intelligence OCR
          </h3>
        </div>
         
        <div
          class="metadata svelte-zccvfh"
        >
          <p
            class="author svelte-zccvfh"
          >
            <a
              class="svelte-zccvfh"
              href="http://github.com/MuckRock/documentcloud-azure-document-intelligence-ocr-addon"
              rel="noopener noreferrer"
              target="_blank"
              title="View Source"
            >
              MuckRock
            </a>
          </p>
           
          <span
            class="badge svelte-zccvfh"
            role="status"
          >
            <div
              class="badge svelte-1o24gyr"
              style="--badge-color: transparent; --label-color: var(--premium);"
            >
              <span
                class="icon svelte-1o24gyr"
              >
                <svg
                  class="svelte-13bl109"
                  height="16px"
                  viewBox="0 0 16 16"
                  width="16px"
                >
                  <title>
                    Premium
                  </title>
                  <path
                    clip-rule="evenodd"
                    d="M7 1C7 3.76142 4.76142 6 2 6C4.76142 6 7 8.23858 7 11C7 8.23858 9.23858 6 12 6C9.23858 6 7 3.76142 7 1Z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M11 9C11 10.6569 9.65685 12 8 12C9.65685 12 11 13.3431 11 15C11 13.3431 12.3431 12 14 12C12.3431 12 11 10.6569 11 9Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
               
              <span
                class="label svelte-1o24gyr"
              >
                Premium
              </span>
            </div>
          </span>
        </div>
      </div>
       
      <div
        class="description svelte-zccvfh"
      >
        <p>
          This Add-On uses Azureâ€™s Document Intelligence API to OCR documents. The document(s) must be public to be processed. This Add-On uses 1 AI Credit per page.
        </p>
      </div>
    </div>
  </a>
</div>
`;
